src/Injector.ts

```diff
@@ -1,154 +1,22 @@
-import { assign } from '@dojo/core/lang';
 import { Evented } from '@dojo/core/Evented';
-import {
-	diffProperty,
-	WidgetBase
-} from './WidgetBase';
-import { decorate, isHNode, isWNode } from './d';
-import { always } from './diff';
-import {
-	Constructor,
-	DNode,
-	HNode,
-	WidgetProperties,
-	WNode
-} from './interfaces';
 
-export interface GetProperties {
-	<P extends WidgetProperties>(inject: any, properties: P): any;
-}
+export class Injector extends Evented {
 
-export interface GetChildren {
-	(inject: any, children: DNode[]): DNode[];
-}
+	private _context: any;
 
-/**
- * The binding mappers for properties and children.
- */
-export interface Mappers {
-	getProperties: GetProperties;
-	getChildren: GetChildren;
-}
-
-/**
- * Default noop Mappers for the container.
- */
-export const defaultMappers: Mappers = {
-	getProperties(inject: any, properties: any): any {
-		return Object.create(null);
-	},
-	getChildren(inject: any, children: DNode[]): DNode[] {
-		return [];
-	}
-};
-
-/**
- * Base context class that extends Evented and
- * returns the context using `.get()`.
- */
-export class Context<T = any> extends Evented {
-
-	private _context: T;
-
-	constructor(context: T = <T> {}) {
+	constructor(context: any) {
```

Comment:
What was the reason/motivation for the context being untyped?  Not challenging, just wondering.

Reply:
There was a challenge with the typing, I'll try it again and let you know (can't remember off the top of my head).

Also wasn't sure what the benefit really was, seen as it's injected at run time there's it didn't really give much to the consumer. The important typing is really on the `getProperties` function that gets passed the context, which the user can type as needed.

Reply:
I was imaging the challenge it seems, updated to accept a generic type.

Reply:
I do get the point about the _value_ of it...